runtime/interpreter/encode.go

```diff
@@ -801,19 +808,34 @@ func (e *Encoder) prepareCompositeValue(
 	interface{},
 	error,
 ) {
-	fields := make(map[string]interface{}, v.Fields.Len())
+	fieldPairs := v.Fields.pairs
+	fieldLen := len(fieldPairs)
+
+	// Sort field names lexicographically.
+	fieldNames := make([]string, fieldLen)
+
+	index := 0
+	for name := range fieldPairs {
+		fieldNames[index] = name
+		index++
+	}
 
-	for pair := v.Fields.Oldest(); pair != nil; pair = pair.Next() {
-		fieldName := pair.Key
-		value := pair.Value
+	sort.Strings(fieldNames)
+
+	// Create fields (key and value) array, sorted by field name.
+	fields := make(cborArray, fieldLen*2)
```

Comment:
I think we can just keep the existing iteration over the ordered map, because that already results in a deterministic encoding. Sorting the fields also works, but is not necessary I think

Reply:
I tried this approach (it's faster and cleaner) but ran into a test error.  It causes this test to fail:

- `TestInterpretCompositeValueFieldEncodingOrder`  in interpreter_test.go.  

I strongly prefer your suggestion and am hoping the test might be obsolete or unneeded.

This is the code block incorporating the faster and cleaner code you suggested:
```go
	fields := make(cborArray, v.Fields.Len()*2)

	i := 0
	for pair := v.Fields.Oldest(); pair != nil; pair = pair.Next() {
		fieldName := pair.Key
		value := pair.Value

		valuePath := append(path[:], fieldName)

		prepared, err := e.prepare(value, valuePath, deferrals)
		if err != nil {
			return nil, err
		}

		fields[i], fields[i+1] = fieldName, prepared
		i += 2
	}

```

I suspect `TestInterpretCompositeValueFieldEncodingOrder` tests if encoding composites with entries of different orderings is deterministic.  

Test expects: 
- CBOR data ["a", 1, "b", 2, "c", 3] by encoding composite with entries `["self.a = 1", "self.b = 2", "self.c = 3"]`.
- CBOR data ["a", 1, "b", 2, "c", 3] by encoding composite with entries `["self.b = 2", "self.c = 3", "self.a = 1", ]`. 

Without sorting, this implementation produces:
- CBOR data ["a", 1, "b", 2, "c", 3] by encoding composite with entries `["self.a = 1", "self.b = 2", "self.c = 3"]`.
- CBOR data ["b", 2, "c", 3, "a", 1] by encoding composite with entries `["self.b = 2", "self.c = 3", "self.a = 1", ]`.

Any thoughts or suggestions?

Reply:
This test checks that no matter what the order of initialization is, the encoding is always the same. That's not required though, we only have to ensure the encoding order is always deterministic, and the ordered map for the fields ensures that.

I've adjusted the test case in f2a268b to prepare the expected encodings for all permutations of initializing the fields